### PR TITLE
chore: update vite to latest

### PR DESCRIPTION
**Pull request by Dunqing**
### Description
Vitest v1.0 has been released and we can update the Vite version to v5.0.0

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -2287,15 +2287,27 @@ packages:
...
+    engines: {node: '>=6.9.0'}
+    dependencies:
+      '@babel/highlight': 7.23.4
+      chalk: 2.4.2
+
   /@babel/compat-data@7.23.2:
     resolution: {integrity: sha512-0S9TQMmDHlqAZ2ITT95irXKfxN9bncq8ZCoJhun3nHL/lLUxd2NKBJYoNGWH7S0hz6fRQwWlAWn/ILM0C70KZQ==}
     engines: {node: '>=6.9.0'}
 
+  /@babel/compat-data@7.23.5:
```
Comment (wojtekmaj) [inline pnpm-lock.yaml:2301]: This PR introduces a lot of duplicate packages, I wonder if that's expected

File: packages/vitest/package.json [hunk 2]
```diff
@@ -155,7 +155,7 @@
     "strip-literal": "^1.3.0",
     "tinybench": "^2.5.1",
     "tinypool": "^0.8.1",
-    "vite": "^5.0.0-beta.19 || ^5.0.0",
+    "vite": "^5.0.5",
```
Comment (sheremet-va) [inline packages/vitest/package.json:158]: Vitest supports Vite ^5.0.0, no need to limit this.

File: packages/vite-node/package.json [hunk 3]
```diff
@@ -87,7 +87,7 @@
     "debug": "^4.3.4",
     "pathe": "^1.1.1",
     "picocolors": "^1.0.0",
-    "vite": "^5.0.0-beta.15 || ^5.0.0"
+    "vite": "^5.0.5"
```
Comment (sheremet-va) [inline packages/vite-node/package.json:90]: ViteNode supports Vite ^5.0.0, no need to limit this.

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (Dunqing) [reply pnpm-lock.yaml]: It seems to have been caused by me accidentally updating all the versions. I revert the lock now.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**